Source/utils/string_variant.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <string>
+#include "utils/stdcompat/string_view.hpp"
+
+namespace devilution {
+
+class StringVariant
+{
+	std::string str;
+	string_view view;
+
+public:
+	StringVariant()
+	{
+	}
+
+	explicit StringVariant(string_view view)
+	    : view(view)
+	{
+	}
+
+	explicit StringVariant(std::string str)
+	    : str(std::move(str))
+	{
+	}
+
+	explicit StringVariant(const char *cstr)
+	    : view(cstr)
+	{
+	}
+
+	explicit StringVariant(char *cstr)
+	    : str(cstr)
+	{
+	}
+
+	StringVariant &operator =(const char *cstr)
```

Comment:
See comment above about factory methods. Can remove all of these assignment operators and just have an assignment operator from `StringVariant`